app/components/publications/editable-publication-status-pill.js

```diff
@@ -75,6 +76,7 @@ export default class PublicationStatusPill extends Component {
       );
       this.loadStatus.perform();
       this.loadDecision.perform();
+      this.router.refresh('publications.publication')
```

Comment:
In case the route refreshes and the component will re-render the constructor will indeed be hit again. 

I still need to have a look in depth at this PR. Refreshing the router from inside a component looks like a risky pattern to introduce at first sight.

Reply:
Agreed that it's quite drastic, but was the only solution I could find (though there might be better ways that I just haven't found).

For context: [this component](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/development/app/components/publications/publication/publication-activities/publications-info-panel.hbs#L59) is show in the "Publicaties" tab and it contains the "Publicatiedatum" field that we want to show the user. Here, the publication date is fetched in task before showing it to the user. When editing the publication status to published via the pill at the top of the page, we want the date to get updated and displayed in this page.

I couldn't find a way to force the task to re-fire (so that the new data gets loaded), so I went for a refresh.

Reply:
"Refreshing the router from inside a component looks like a risky pattern to introduce at first sight."
I get what you mean, if this component is reused in some other route this would not work.
Passing a controller method down to do the component and call that with `this.args.refreshRoute` seems like a better option. Then we do the refreshing in the controller of the route we are in.